### PR TITLE
Update deploy script to use variable FLASHSYSTEM_DRIVER_FULL_IMAGE_NAME in file manager_config_patch.yaml

### PR DESCRIPTION
**Pull request by lbscorpio**
Please help to review, thanks.

### CODE REVIEW

File: hack/deploy-operator.sh [hunk 1]
```diff
@@ -23,6 +23,12 @@ source hack/common.sh
 echo "Deploying the operator in the cluster..."
 
 pushd config/manager
+if [ "$LOCAL_OS_TYPE" == "Darwin" ] && [[ "$(sed --version 2>&1 | head -n 1 | awk -F " " '{print $2}')" == "illegal" ]]; then
```
Comment (iamniting) [inline hack/deploy-operator.sh:26]: can not we use `kustomize` here something like below

`cd config/manager && kustomize edit set image controller=${IMG}`

**Review by iamniting — CHANGES_REQUESTED** [review]
I would really encourage you to wrap the commit-msg to 72 characters.

e.g. https://github.com/red-hat-data-services/odf-operator/blob/main/CONTRIBUTING.md

File: hack/deploy-operator.sh (hunk 1 above)
Comment (lbscorpio) [reply hack/deploy-operator.sh:26]: The line 32 of the same file already have `kustomize edit set image controller=${IMG}`, which is for controller image(operator here). The one I updated is the environment variable value, I am not familiar with kustomize to update it, so use sed method.

Comment (lbscorpio) [reply hack/deploy-operator.sh:26]: Checked the `kustomize edit set` command help usage, only 5 commands(image/nameprefix/namespace/namesuffix/replicas) value can be updated, which is not fulfilled my request. Fyi, thanks.
```
$ kustomize edit set -h
Sets the value of different fields in kustomization file.

Usage:
  kustomize edit set [command]

Examples:

	# Sets the nameprefix field
	kustomize edit set nameprefix <prefix-value>

	# Sets the namesuffix field
	kustomize edit set namesuffix <suffix-value>


Available Commands:
  image       Sets images and their new names, new tags or digests in the kustomization file
  nameprefix  Sets the value of the namePrefix field in the kustomization file.
  namespace   Sets the value of the namespace field in the kustomization file
  namesuffix  Sets the value of the nameSuffix field in the kustomization file.
  replicas    Sets replicas count for resources in the kustomization file

Flags:
  -h, --help   help for set

Global Flags:
      --stack-trace   print a stack-trace on error

Use "kustomize edit set [command] --help" for more information about a command.
```

**Comment by lbscorpio** [comment]
@iamniting I have updated the commit-msg, thanks for your input.

**Review by iamniting — APPROVED** [review]

**Review by gongyj-ibm — APPROVED** [review]
lgtm, thanks

**Pull request merged**